### PR TITLE
Build iconographer as a tarball

### PR DESCRIPTION
**Pull request by GabiThume**
- [x] Create Makefile to compile iconographer using imgflo-dependencies
- [x] Deploy on Travis
- [x] Deploy on Heroku

### CODE REVIEW

**Comment by automata** [comment]
It's also being built/deployed from Heroku too: https://travis-ci.org/the-grid/iconographer/builds/194928441#L5579.

File: upload-to-s3.sh [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+if [ -z "$TARGET" ];
+then
+  echo "TARGET not set, going with system name";
+  TARGET=`uname -n`
+fi
+
+if [ -z "$AMAZON_API_TOKEN" ];
+then
+  echo "Amazon API Token not provided, skipping upload";
```
Comment (jonnor) [inline upload-to-s3.sh:39]: Why not fail hard? If you don't want to upload something, probably should not call an upload script?

File: Makefile [hunk 2]
```diff
@@ -1,4 +1,51 @@
...
-include .mm/bin
-BIN_PKGMODULES=gegl-0.3
+PREFIX=$(shell echo `pwd`)
+PREFIX_DEP=$(PREFIX)/build/imgflo-dependencies/install
+VERSION=$(shell echo `git describe --tags`)
+CFLAGS += -Wall -Wextra
+CFLAGS += -Ofast -g -rdynamic
+CFLAGS += -fPIC
+CFLAGS += -fomit-frame-pointer
+CFLAGS += -funroll-loops -fforce-addr -ftracer -fpeel-loops -fmerge-all-constants
```
Comment (jonnor) [inline Makefile:8]: Where do these CFLAGS come from and why are they important?

Comment (jonnor) [reply Makefile:8]: Nevermind, I see they were in the original Makefiles. I guess we keep them then.

File: .travis.yml [hunk 3]
```diff
@@ -0,0 +1,34 @@
+language: c
```
Comment (jonnor) [inline .travis.yml:1]: Yay continious integration

File: .travis.yml [hunk 4]
```diff
@@ -0,0 +1,34 @@
...
+    - libsdl1.2-dev
+    - libjson-glib-dev
+    - libglib2.0-dev
+    - libgif-dev
+    - libpango1.0-dev
+    - librsvg2-dev
+    - libsdl1.2-dev
+    - openexr
+script:
+- make release-and-upload VERSION=`git describe --tags` TARGET=travis-${TRAVIS_OS_NAME}
```
Comment (jonnor) [inline .travis.yml]: This should call a `check` or `release-check` target. Upload should be separate.

**Review by jonnor — CHANGES_REQUESTED** [review]
Should change upload logic to be only on deploy.

**Comment by automata** [comment]
@jonnor thanks for the suggestions. Can you take a look again, pls? I'll remove 8b85e40 before merging, that was just to conveniently check deploy from this PR.

**Review by jonnor — APPROVED** [review]

**Pull request merged**